src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -0,0 +1,258 @@
+        });
+    }
+
+    /**
+     * Used to calculate the Context Menu Dimensions
+     *
+     * @returns {JSX}
+     * @memberof PopoverReportActionContextMenu
+     */
+    measureContent() {
+        return (
+            <ReportActionContextMenu
+                isVisible
+                selection={this.state.selection}
+                reportID={this.state.reportID}
+                reportAction={this.state.reportAction}
+                hidePopover={this.hideContextMenu}
+                showDeleteConfirmModal={this.showDeleteConfirmModal}
+            />
+        );
+    }
+
+    confirmDeleteAndHideModal() {
+        deleteReportComment(this.state.reportID, this.state.reportAction);
+        this.setState({isDeleteCommentConfirmModalVisible: false});
+    }
+
+    hideDeleteConfirmModal() {
+        this.setState({
+            reportID: 0,
+            reportAction: {},
+            isDeleteCommentConfirmModalVisible: false,
+        });
+    }
+
+    /**
+     * Opens the Confirm delete action modal
+     * @param {Number} reportID
+     * @param {Object} reportAction
+     * @memberof PopoverReportActionContextMenu
+     */
+    showDeleteConfirmModal(reportID, reportAction) {
+        this.setState({reportID, reportAction, isDeleteCommentConfirmModalVisible: true});
+    }
+
+    render() {
+        return (
+            <>
+                <PopoverWithMeasuredContent
+                    isVisible={this.state.isPopoverVisible}
+                    onClose={this.hideContextMenu}
+                    onModalHide={this.contextMenuHidden}
+                    anchorPosition={this.state.popoverAnchorPosition}
+                    animationIn="fadeIn"
+                    animationOutTiming={1}
+                    measureContent={this.measureContent}
+                    shouldSetModalVisibility={false}
+                    fullscreen={false}
+                >
+                    <ReportActionContextMenu
+                        isVisible
+                        reportID={this.state.reportID}
+                        reportAction={this.state.reportAction}
+                        draftMessage={this.state.reportActionDraftMessage}
+                        hidePopover={this.hideContextMenu}
+                        showDeleteConfirmModal={this.showDeleteConfirmModal}
+                    />
+                </PopoverWithMeasuredContent>
+                <ConfirmModal
+                    title={this.props.translate('reportActionContextMenu.deleteComment')}
+                    isVisible={this.state.isDeleteCommentConfirmModalVisible}
+                    onConfirm={this.confirmDeleteAndHideModal}
+                    onCancel={this.hideDeleteConfirmModal}
+                    prompt={this.props.translate('reportActionContextMenu.deleteConfirmation')}
+                    confirmText={this.props.translate('common.delete')}
+                    cancelText={this.props.translate('common.cancel')}
+                />
```

Comment:
Modal is an expensive component. Its better to move it along with ContextMenu if we are moving the ContextMenu itself. Both ContextMenu and delete confirm Modal was used in a similar way so I moved this along with contextmenu.
After Moving the actions into and external libs it makes more sense.

Reply:
Right, I'm not suggesting we include it in every report action item and it's good that it's now isolated instead of created `n` times. But it feels out of place here and can sit next to the this component instead of inside it.